pkg/apis/triggers/v1alpha1/trigger_template_validation.go

```diff
@@ -91,16 +91,24 @@ func verifyParamDeclarations(params []ParamSpec, templates []TriggerResourceTemp
 		declaredParamNames[param.Name] = struct{}{}
 	}
 	for i, template := range templates {
-		// Get all params in the template $(params.NAME)
+		// Get all params in the template $(tt.params.NAME) or $(params.NAME)
 		templateParams := paramsRegexp.FindAllSubmatch(template.RawExtension.Raw, -1)
 		for _, templateParam := range templateParams {
-			templateParamName := string(templateParam[1])
+			templateParamName := string(templateParam[2])
```

Comment:
I realize this is following what was set before, but the direct array access is at risk for nil panics. I think we should do 2 things:

1) Have a length check prior to accessing `templateParam[2]`.
2) Create a more exhaustive set of regexp checks instead of the more e2e tests we see here. (mainly to give us more confidence that the regexp is matching to what we expect, even in unexpected cases).

Reply:
Actually 
```
templateParams := paramsRegexp.FindAllSubmatch(template.RawExtension.Raw, -1)
```
templateParams length will be 0 in any unexpected cases because of Regular Expression
```
regexp.MustCompile(`\$\((params|tt.params).([_a-zA-Z][_a-zA-Z0-9.-]*)\)`)
```
If user provides any such values like below
```
1. params1.foo
2. tt1.params.foo
3. tt.params1.foo
4. tt.foo
5. .foo
```
the `templateParams` itself is `0` and because of that it never enters into for loop 

But yes need to add more test cases to ensures the regex will not break.

Reply:
Added test cases to cover few more scenario